buster/documents_manager/base.py

```diff
@@ -107,6 +108,75 @@ def add(
 
         self._add_documents(df, **add_kwargs)
 
+    def batch_add(
+        self,
+        df: pd.DataFrame,
+        batch_size: int = 3000,
+        min_time_interval: int = 60,
+        num_workers: int = 16,
+        embedding_fn: callable = get_embedding_openai,
+        csv_checkpoint: Optional[str] = None,
+        **add_kwargs,
+    ):
+        """
+        This function takes a DataFrame and adds its data to a DataManager instance in batches.
+        It ensures that a minimum time interval is maintained between successive batches
+        to prevent timeouts or excessive load. This is useful for APIs like openAI with rate limits.
+
+        Args:
+            df (pandas.DataFrame): The input DataFrame containing data to be added.
+            batch_size (int, optional): The size of each batch. Defaults to 3000.
+            min_time_interval (int, optional): The minimum time interval (in seconds) between batches.
+                                            Defaults to 60.
+            num_workers (int, optional): The number of parallel workers to use when adding data.
+                                        Defaults to 32.
+            embedding_fn (callable, optional): A function that computes embeddings for a given input string.
+                Default is 'get_embedding_openai' which uses the text-embedding-ada-002 model.
+            csv_checkpoint: (str, optional) = Path to save a copy of the dataframe with computed embeddings for later use.
+                Omit the .csv extension. It will be appended with batch_idx number to separate between batches.
```

Comment:
should we merge them all in one file?

Reply:
we could after the fact, the idea is that we have temporary checkpoints as time goes on should anything go wrong.

Another option is to have a file that we open, append to, and close every time

Reply:
updated to append to an existing file if present